### ISSUE TITLE
Too many redirects after login

### ISSUE THREAD

**Issue by bbunlock** [trntv/yii2-starter-kit#530]
HI, first thanks for the great work putting all this together, I installed it on my local machine (xampp) a few weeks ago, I initially had a few problems getting it to work in single domain but got it sorted with a few .htaccess tweaks provided by people having similar issues (google search is your friend).

I then spent the next 4 weeks working on the project and finally got it finished yesterday, I uploaded to my host and it all seemed to be working fine, that is until I login, once I login I get too many redirects on any page I try and visit including the standard pages but nothing works.

every page times out with to many redirects, if I clear my cookies I can see the standard pages again (home, about, contact etc) but nothing works after I login.

can anyone give some advice on how I might get this working, I have tried many versions of the .htaccess but nothing works after I login.

any help or suggestions would be grateful.

Kind regards

Wayne

**Comment by bbunlock** [comment]
just an update, I have been trying countless fixes that I have seen on the net about setting this up on a single domain but so far nothing works, I have however found a little out.

1, if I goto mysite.com/backend/web I get the login screen for the backend, if I login I get redirected to the front page where I am logged out, I need to login separate for the front section.

If I try to go anywhere in the backend I am just redirected to the sites home page, I can not see any part of the backend even though I am logged in.

2, If I login to the frontend I get the redirect loop I mentioned above BUT if I login and do not tick the remember me box I can see the frontend just fine, all the pages work as they should, but of course this means I am only logged in for this session.

Hope the info helps with some ideas as to what is happening.

Kind regards

Wayne

**Comment by XzAeRo** [comment]
I had this problem before. Here's a checklist you can go through to verify some things:

- Clean all your application cookies, or test from incognito mode.
- Check if the application is generating a login cookie using Chrome or Firefox developer tools.
- Check if the application login cookie is using the correct domain (it should match your application url, eg: http://yourapp.com).
  - If not, check why it isn't.
  - Hint: check your `.env` file in the URL settings.
- Check that you have setup the RBAC settings (by running `console/yii rbac-migrate/up`).
- Check that you're verifying the user permissions before trying to run the login procedure.

Hope this helps!

**Comment by trntv** [comment]
something is wrong with your ``baseUrl``. Try to update starter kit to latest version and change ``.env`` file HOST_INFO and BASE_URL variables to match your installation

**Comment by davidjeddy** [comment]
I too have run into this. Pulled master branch, single domain configuration, updated .env to reflect new ENV keys/value pairs, rebuild docker containers, testing in incognito mode in Chrome.

All frontend functionality works as expected.

When logging into the backend the sign-in/login action redirects back to itself when logging in with correct credentials. I think it has something to do with the following part of backend controller SignInController.

```
        if ($model->load(Yii::$app->request->post()) && $model->login()) {
            return $this->goBack();
        } else {
            return $this->render('login', [
                'model' => $model
            ]);
        }
```

When logging in there is not URI to `goBack()` to.

**Comment by davidjeddy** [comment]
I am having this issue as well. It's something configuration related but can not figure out what. See attached .env

```
# Framework
# ---------
YII_DEBUG   = true
YII_ENV     = dev

# Application
# -----------
LINK_ASSETS=true

# Databases
# ---------
DB_DSN           = {redacted}
DB_USERNAME      = pieces
DB_PASSWORD      = pieces
DB_TABLE_PREFIX  =

TEST_DB_DSN           = {redacted}
TEST_DB_USERNAME      = root
TEST_DB_PASSWORD      = root
TEST_DB_TABLE_PREFIX  =

# Urls
# ----
FRONTEND_HOST_INFO    = http://pieces-compendium.dev
FRONTEND_BASE_URL     = /
BACKEND_HOST_INFO     = http://pieces-compendium.dev
BACKEND_BASE_URL      = /backend/web
STORAGE_HOST_INFO     = http://pieces-compendium.dev
STORAGE_HOST_INFO     = /storage/web

# Other
# -----
SMTP_HOST = mailcatcher
SMTP_PORT = 1025

FRONTEND_COOKIE_VALIDATION_KEY = {redacted}
BACKEND_COOKIE_VALIDATION_KEY  = {redacted}

ADMIN_EMAIL    = admin@{redacted}.com
ROBOT_EMAIL    = robot@{redacted}.com

GITHUB_CLIENT_ID     = NULL
GITHUB_CLIENT_SECRET = NULL

GLIDE_SIGN_KEY       = {redacted}
GLIDE_MAX_IMAGE_SIZE = 4000000
```

Login works, but am redirected back to the login view on successful submission.

**Comment by davidjeddy** [comment]
@bbunlock I think I figured out the problem. Using the new `FRONTEND_HOST_INFO` the `FRONTEND_BASE_URL` has to also be specified. Without both items in the .env things to not work right.

@trntv We may want to migration both sub and single domain configurations to use the same .env; or have two separate .env.dist examples. It is becoming confusing for some how and where to make the appropriate edits for the different configurations.

**Comment by Alofoxx** [comment]
This is happening to me on a new install and I'm not sure why
Using single domain instuctions.

**Comment by alfredleo** [comment]
I once faced almost the same problem with yii2 project. 
error was: Request exceeded the limit of 10 internal redirects due to probable configuration error.
So the fix I came with `RewriteBase /` in .htaccess file.

**Comment by imranswe** [comment]
For me removing a property `baseUrl` from `'components' => [ 'request' => ['cookieValidationKey' => env('BACKEND_COOKIE_VALIDATION_KEY'),'baseUrl' => env('BACKEND_HOST_INFO')]]` in file `backend/config/web.php`

This worked because I have created three virtual hosts like: api.mydomain.development, backend.mydomain.development, frontend.mydomain.development and I did not need any baseUrl in the config. I guess this baseUrl is used to specify route groups like /admin. In my case both my backend host url and this baseUrl were same and both were full URLs and hence was creating redirect loop causing the too many redirects error.
